src/test/java/test/schuraytz/RendererTest.java

```diff
@@ -0,0 +1,84 @@
+package test.schuraytz;
+
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.*;
+import com.badlogic.gdx.physics.box2d.Body;
+import com.badlogic.gdx.physics.box2d.BodyDef;
+import com.badlogic.gdx.physics.box2d.CircleShape;
+import com.badlogic.gdx.physics.box2d.FixtureDef;
+import com.badlogic.gdx.physics.box2d.World;
+import com.badlogic.gdx.utils.Array;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import static org.mockito.Mockito.*;
+
+import pinball.Renderer;
+import java.awt.Graphics2D;
+
+public class RendererTest {
+
+    @Test
+    public void render() {
+        //given
+        World mWorld = mock(World.class);      // Can only mock a final class once manually activate the extension option -  https://antonioleiva.com/mockito-2-kotlin/
+        Renderer renderer = new Renderer(mWorld, 10);
+
+        CircleShape mShape = mock(CircleShape.class);
+        FixtureDef mFixtureDef = mock(FixtureDef.class);
+        mFixtureDef.shape = mShape;
+        Body mBody = mock(Body.class);
+        Graphics2D mGraphic = mock(Graphics2D.class);
+
+        //when
+        renderer.render(mGraphic);
+
+        //then
+        doAnswer((i) -> {
+            Array<Body> bodies = new Array<>();
+            bodies.add(mBody);
+            return null;
+        }).when(mWorld.getBodies(Array.class));
+
+        doReturn(new Vector2(10, 10)).when(mBody).getPosition();
```

Comment:
These need to be in the `given` part of the test. 
You are setting up how the mocks should respond when methods are called on them. This needs to happen before you call `renderer.render()`